pkg/gitops/repository.go

```diff
@@ -28,24 +28,23 @@ var _ repositorier = (*repository)(nil)
 type repository struct {
 	gh          gitProvider
 	remote      RemoteConfig
-	sshKey      sshKeyer
 	tmpRepoPath string
 }
 
 // RemoteConfig is a git remote configuration.
 type RemoteConfig struct {
-	URL, Branch, Folder string
+	Repo   *githubRepo
```

Comment:
If we store a reference to the repo itself, do we still need the `gitProvider` interface? It's unclear to me at this stage where exactly the responsibilities of `repository`, `github`, and `githubRepo` lie

Reply:
Yeah, I agree. It got a bit messy :/ trying to make more sense of them

Reply:
Pushed a refactor, please check it out when you have the time! :)

Reply:
Much better, thanks!